src/value.rs

```diff
@@ -27,14 +29,21 @@ pub struct AllocatedValue {
 #[derive(Copy, Clone, Debug)]
 pub struct AllocatedQuantity {
     pub variable: Variable,
-    pub assignment: Option<u64>,
+    pub assignment: Option<SignedInteger>,
+}
+
+/// Represents a signed integer in the range [-(2^64-1) .. 2^64-1]
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
```

Comment:
I wouldn't derive Eq/PartialEq unless we really need them, and if we do, we need to be explicit on whether we allow Negative(0) == Positive(0) or not. Hopefully we don't allow these equality checks at all and do not have to worry about this.

Reply:
We need to derive Eq and PartialEq since Value derives Eq and PartialEq and right now, Value contains a `SignedInteger` for the quantity?

Reply:
Hmmm.... Ok, lets derive and not allow `Negative(0) == Positive(0)` for the reason that these are different _types_. Negative value is a counterbalance for the borrowed positive value and is a distinct type in that it's _portable_ in ZkVM terms.

Reply:
derive(PartialEq) would return false on different variants, so we don't have to do anything extra